classes/Link.php

```diff
@@ -503,25 +507,25 @@ public function getAdminLink($controller, $with_token = true, $sfRouteParams = a
                 if (!$redirectLegacy) {
                     if (array_key_exists('id_product', $sfRouteParams)) {
                         if (array_key_exists('deleteproduct', $sfRouteParams)) {
-                            return $this->getBaseLink().basename(_PS_ADMIN_DIR_).'/product/unit/delete/' . $sfRouteParams['id_product'];
+                            return $this->getBaseLink().basename(_PS_ADMIN_DIR_).$index.'/product/unit/delete/'.$sfRouteParams['id_product'];
```

Comment:
I don't think `Link class now depends on Symfony router` is a bad thing.

If one of its routes is updated, we won't have to change anything here if the URLs are not hardcoded anymore